# Ex1 - Filtrar y ordenar datos

### Paso 1. Importar las bibliotecas necesarias

In [2]:
import pandas as pd
import numpy as np

### Paso 2. Importar el conjunto de datos de esta [dirección](https://raw.githubusercontent.com/justmarkham/DAT8/master/data/chipotle.tsv). 

In [3]:
chipo=pd.read_csv(r'https://raw.githubusercontent.com/justmarkham/DAT8/master/data/chipotle.tsv',delimiter='\t')

### Paso 3. Asígnalo a una variable llamada chipo.

In [4]:
chipo.head(20)

order_id  quantity                              item_name  \
0          1         1           Chips and Fresh Tomato Salsa   
1          1         1                                   Izze   
2          1         1                       Nantucket Nectar   
3          1         1  Chips and Tomatillo-Green Chili Salsa   
4          2         2                           Chicken Bowl   
5          3         1                           Chicken Bowl   
6          3         1                          Side of Chips   
7          4         1                          Steak Burrito   
8          4         1                       Steak Soft Tacos   
9          5         1                          Steak Burrito   
10         5         1                    Chips and Guacamole   
11         6         1                   Chicken Crispy Tacos   
12         6         1                     Chicken Soft Tacos   
13         7         1                           Chicken Bowl   
14         7         1                    Chips and Guacamole   
15         8         1  Chips and Tomatillo-Green Chili Salsa   
16         8         1                        Chicken Burrito   
17         9         1                        Chicken Burrito   
18         9         2                            Canned Soda   
19        10         1                           Chicken Bowl   

                                   choice_description item_price  
0                                                 NaN     $2.39   
1                                        [Clementine]     $3.39   
2                                             [Apple]     $3.39   
3                                                 NaN     $2.39   
4   [Tomatillo-Red Chili Salsa (Hot), [Black Beans...    $16.98   
5   [Fresh Tomato Salsa (Mild), [Rice, Cheese, Sou...    $10.98   
6                                                 NaN     $1.69   
7   [Tomatillo Red Chili Salsa, [Fajita Vegetables...    $11.75   
8   [Tomatillo Green Chili Salsa, [Pinto Beans, Ch...     $9.25   
9   [Fresh Tomato Salsa, [Rice, Black Beans, Pinto...     $9.25   
10                                                NaN     $4.45   
11  [Roasted Chili Corn Salsa, [Fajita Vegetables,...     $8.75   
12  [Roasted Chili Corn Salsa, [Rice, Black Beans,...     $8.75   
13  [Fresh Tomato Salsa, [Fajita Vegetables, Rice,...    $11.25   
14                                                NaN     $4.45   
15                                                NaN     $2.39   
16  [Tomatillo-Green Chili Salsa (Medium), [Pinto ...     $8.49   
17  [Fresh Tomato Salsa (Mild), [Black Beans, Rice...     $8.49   
18                                           [Sprite]     $2.18   
19  [Tomatillo Red Chili Salsa, [Fajita Vegetables...     $8.75

# Nombre del producto de valor máximo

In [8]:
chipo['item_price'].max()

'$9.39 '

- Parece que mi columna item_price no es del tipo int ya que el valor max no es $9.39 

In [9]:
chipo.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4622 entries, 0 to 4621
Data columns (total 5 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   order_id            4622 non-null   int64 
 1   quantity            4622 non-null   int64 
 2   item_name           4622 non-null   object
 3   choice_description  3376 non-null   object
 4   item_price          4622 non-null   object
dtypes: int64(2), object(3)
memory usage: 180.7+ KB


- Observamos que la columna item_price es del tipo object y que no tiene null.

- Miremos que tipos tiene la columna item_price

In [16]:
# la columna no tiene Null 
#chipo[chipo['item_price'].isna()]

chipo['item_price'].apply(type)

0       <class 'str'>
1       <class 'str'>
2       <class 'str'>
3       <class 'str'>
4       <class 'str'>
            ...      
4617    <class 'str'>
4618    <class 'str'>
4619    <class 'str'>
4620    <class 'str'>
4621    <class 'str'>
Name: item_price, Length: 4622, dtype: object

- vemos que la clase es de tipo string 
- vamos a eliminar el simbolo $ u espacios o comas que pueda tener y convertirlo en folat

In [17]:
chipo['item_price'] = chipo['item_price'].replace({'\$': '', ',': ''}, regex=True).astype(float)

- volvamos a comprobar de que tipo es nuestra columna 

In [18]:
chipo['item_price'].dtype

dtype('float64')

- parece que ahora si es del tipo float 
- vamos a sacar el maximo comprobando que el numero de la columna quantity sea 1 ,
ya que nos interesa que el valor unitario del producto.

In [26]:
filtrado_1=chipo[chipo['quantity']==1]

precio_max=filtrado_1['item_price'].max()

precio_max

np.float64(11.89)

- como puede haber mas de un producto que sea maximo 
- vamos a ver cuantos hay y quienes son 

### Paso 4. ¿Cuántos productos cuestan más de 10,00 dólares?

### Paso 5. ¿Cuál es el precio de cada artículo?
###### imprimir un marco de datos con sólo dos columnas nombre_artículo y precio_artículo

### Paso 6. Ordenar por el nombre del artículo

### Paso 7. ¿Cuál fue la cantidad pedida del artículo más caro? 2 formas

### Paso 8. ¿Cuántas veces pidió ensalada de verduras?

### Paso 9. ¿Cuántas veces pidió alguien más de un refresco en lata?